ChatSpace: Front

BayG opened the pull request:
# What
Mounted several features in a display of chatting via index.html.haml and message.scss
# Why
For determining the layout of chat screen

BayG committed:
ChatSpace: Front

seoky0 commented on app/assets/javascripts/messages.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
使用しないファイルなので消しておきましょう

seoky0 commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,13 @@
+@import "./mixin/clearfix";
+@import "messages";
+@import "./mixin/flexiblebox";
+/*@import "./modules/sidebar";
```
不要なコメントアウトであれば消しておきましょう

seoky0 commented on app/helpers/messages_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module MessagesHelper
```
不要なファイルは消しましょう

seoky0 commented on test/controllers/messages_controller_test.rb:
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
テストファイルも不要なので消しておきましょう

seoky0 commented on test/models/message_test.rb:
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
同様に不要ですね

seoky0 commented:
作成したビューのフロントの画像をこちらに貼り付けてください。
<img width="1920" alt="img_chatspace-front_20170301" src="https://cloud.githubusercontent.com/assets/21117182/23445341/1b54eefa-fe7e-11e6-9fb6-14f9111ddacc.png">

BayG committed:
ChatSpace-Front: File organization, etc.

BayG committed:
Editing the title name in application.html.haml (line 5)

BayG commented:
# What
File organization
 - Deleted app/assets/javascripts/messages.coffee
 - Eliminated comment-out from app/assets/stylesheets/application.scss
 - Deleted app/helpers/messages_helper.rb
 - Deleted test/controllers/messages_controller_test.rb
 - Deleted test/models/message_test.rb
# Why
Doesn't need to remain still for mounting ChatSpace features later.

tetsu8dayo commented on Gemfile:
```diff
@@ -49,5 +49,8 @@ group :development do
+
+  # font-awesome
+  gem "font-awesome-rails"
 end
```
haml-rails や font-awesome-rails はdevelopment 環境でしか使わないでしょうか、、、？
そこも含めて、だいたいでいいのでどのgemがどういう機能で、どの環境に入れるべきなのか(development, test, production)考えてまとめて見てください

BayG replied:
理解しました。これはつまり、この二つをDevelpmentのlistに入れずに、Gemfile最上段に挿入すべきということですね。そうすれば、development, test, production ３つの開発環境に組み込められる。

tetsu8dayo commented on test/fixtures/messages.yml:
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
factoryを使うのでこのファイルはいりません
genarator の設定をいじってモデル生成時にこのファイルは作られないようにしてください。

tetsu8dayo replied:
https://railsguides.jp/generators.html
これにいろいろ書いてあるので読んで下さい。

BayG replied:
要件外のファイルを外すというカリキュラムに従い、application.rbにおいて、class Applicationに以下を挿入しました。
```
module ChatSpace
  class Application < Rails::Application
    config.generators do |g|
      g.test       false
      g.coffee     false
      g.helper     false
    end
  end
end
```

narukami894 replied:
このファイルも消してあげましょう！

BayG replied:
消します！

BayG committed:
Edited Gemfile - enabling three gems for three each environment

BayG committed:
Edited application.rb for setting generator: Disabled three files generation.

narukami894 commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,8 @@
+/*
+ *= require font-awesome
```
font-awesomeも@importしてあげましょう！

BayG replied:
```
@import "font-awesome";
```
こうですね！

narukami894 commented on app/assets/stylesheets/config/_variable.scss:
```diff
@@ -0,0 +1,12 @@
+$titleBlack: #333333;
+$textGray: #999999;
+$textBlack: #434A54;
+$footerGray: #dddddd;
```
good

BayG replied:
どうもです！

narukami894 commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,204 @@
+// Place all the styles related to the messages controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
この三行消してあげて大丈夫です！

BayG replied:
了解です！

narukami894 commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,204 @@
+          font-size: 20px;
+          color: $titleBlack;
+          display: inline-block;
+          letter-spacing: 2px
```
; をつけてください！

BayG replied:
おお！さすがです

narukami894 commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,204 @@
+    }
+  }
+}
+
```
最終行の改行はひとつだけにしましょう

BayG replied:
なるほど、了解です！

narukami894 commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,204 @@
+        top: 32px;
+        right: 130px;
+        max-width: 30px;
+
```
改行やインデントがまちまちなので、統一してください！

BayG replied:
気をつけます！

narukami894 commented on app/views/layouts/application.html.haml:
```diff
@@ -2,7 +2,7 @@
 %html
   %head
     %meta{:content => "text/html; charset=UTF-8", "http-equiv" => "Content-Type"}/
```
ハッシュロケットが残っているので、ハッシュロケットを使わない形に書き換えましょう

BayG replied:
指摘どうもです！

BayG replied:
```
content: "text/html; charset=UTF-8", "http-equiv": "Content-Type"
```
こうですか？

narukami894 commented on app/views/layouts/application.html.haml:
```diff
@@ -2,7 +2,7 @@
-    %title ChatSpace
+    %title [WIP]ChatSpace
     = stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true
     = javascript_include_tag 'application', 'data-turbolinks-track' => true
```
こちらもですね

BayG replied:
直します！ｗ

narukami894 commented on config/routes.rb:
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  get 'messages' => 'messages#index'
```
resourcesを使って定義してあげてください！

BayG replied:
以下のように直しました！
```routes.rb
Rails.application.routes.draw do
  root      'messages#index'
  resources :messages, only: [:index, :new, :create]
end
```

BayG committed:
Deleted test/fixtures/messages.yml

Doesn't need to success work because will use "factory".

BayG committed:
Edited text much richer

BayG commented on config/routes.rb:
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  get 'messages' => 'messages#index'
```
”RESTfulなURLを自動生成”でいいですか？

BayG replied:
```
resources :リソース名 [, オプション]
```

BayG committed:
Edited applicaton.scss

- Deleted comment-out "require font-awesome", then
- Added "@import "font-awesome"; in line 7.

BayG committed:
Edited messages.scss

 - Deleted comment-out from line 1 to line 4
 - Added semi-colon at line 101, aside
``` letter-spacing: 2px ```
 - Deleted mean-less line on line 179 and line 204.

BayG committed:
Edited application.html.haml

Converted old-type hash notation to new one, which isn't used
hash-rocket "=>"

BayG committed:
Edited README.md for enabling markdown.

BayG committed:
Edited README.md

 - Removed "###" which was aside of "1. Table" on line 30.
 - Deleted else that doesn't need.

BayG committed:
Restored the text written about CoffeeScript

BayG committed:
Edited routes.rb for making routes easier and securer

BayG commented:
# What
 - Deleted test/fixtures/messages.yml
 - Edited several files
 - Edited README.md (become more readable)

# Why
 - Convenience for referrencing.
 - For making routing easier and securer.

kenchan0503 commented on db/migrate/20170223105312_create_messages.rb:
```diff
@@ -0,0 +1,11 @@
+      t.text :body
+      t.string :image
+      t.integer :user_id
+      t.integer :group_id
```
こちらpreference型にしましょう。
外部キー制約もお願いします！

BayG replied:
reference型ですね！

BayG replied:
```db/migrate/20170223105312_create_messages.rb
create_table :messages do |t|
      t.text :body
      t.string :image
      t.references :user, foreign_key: true
      t.references :group, foreign_key: true
      t.timestamps
```
確認お願いします！

kenchan0503 commented on app/assets/stylesheets/config/_variable.scss:
```diff
@@ -0,0 +1,12 @@
+$titleBlack: #333333;
+$textGray: #999999;
+$textBlack: #434A54;
+$footerGray: #dddddd;
```
#dddddd のように同じ文字が続いてるものは #dddにするとよりgoodです

BayG committed:
Edited two things on db/migrate/20170223105312_create_messages.rb

 - Used the t.references form at line 6 and line 7.
 - Implemented Foreign Key information at line 6 and line 7.

BayG committed:
Eased expression of color HEX code
